src/test/java/io/kestra/plugin/weaviate/QueryTest.java

```diff
@@ -0,0 +1,138 @@
+package io.kestra.plugin.weaviate;
+
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.runners.RunContextFactory;
+import io.kestra.core.serializers.JacksonMapper;
+import io.kestra.core.storages.StorageInterface;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import jakarta.inject.Inject;
+import org.apache.commons.io.Charsets;
+import org.apache.commons.io.IOUtils;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+@MicronautTest
+public class QueryTest {
+
+    public static final String SCHEME = "http";
+    public static final String HOST = "localhost:8080";
+    private static final String QUERY = """
+                       {
+  Get {
+    QueryTest (
+      limit: 50
+    ) {
+      title
+    }
+  }
+}
+                       """;
+
+    @Inject
+    private RunContextFactory runContextFactory;
+
+    @Inject
+    private StorageInterface storageInterface;
+
+    @Test
+    public void testQuery_Without_Internal_Storage() throws Exception {
```

Comment:
You should split your tests by Task ie. BatchTest, QueryTest & DeleteTest
It doesn't prevent you from using Batch task in QueryTest & DeleteTest to populate data but it brings clearer tests.
Also, you can use inheritance to have a base class with all utilities (like url, scheme, ...) and share them among all test classes